ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/eea/EeaSendRawTransaction.java

```diff
@@ -148,15 +116,28 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
     }
   }
 
+  Optional<PrivacyGroup> findPrivacyGroup(
+      final PrivacyController privacyController,
+      final Optional<Bytes> maybePrivacyGroupId,
+      final String enclavePublicKey,
+      final PrivateTransaction privateTransaction) {
+    final Optional<PrivacyGroup> maybePrivacyGroup =
+        findOffchainPrivacyGroup(privacyController, maybePrivacyGroupId, enclavePublicKey);
+    return maybePrivacyGroup;
+  }
+
   JsonRpcResponse createPMTAndAddToTxPool(
```

Comment:
I don't like the fact that this method is doing two things. We can probably break it into two.

Reply:
Done!